### PR TITLE
BUG: Fix reference count issues

### PR DESCRIPTION
**Pull request by seberg**
The new PyInt_FromSsize_t reference was not decref'ed in general.

---

I was hacking on numpy using `pytest-leaks` and ran into a reference issues within cython (numpy.random). This is the on that I managed to track down for now. There is another coming up (although most likely I will only open an issue).

I suppose these should get tests? Just ping me, and I will look into it.

### CODE REVIEW

**Comment by seberg** [comment]
Sorry for the many forced pushs (did not run tests, which I guess would have told me that my code looked weird) and adding of more commits as I am on to a few other things. If you want me to split it up or do style fixes, etc. I am happy to do that. Until then, may push more small things as I track them down.

**Comment by seberg** [comment]
OK, this together with gh-3023 makes the numpy random tests run through cleanly with respect to refcounting issues, so for me this has settled already.

File: Cython/Utility/ObjectHandling.c [hunk 1]
```diff
@@ -497,8 +500,12 @@ static CYTHON_INLINE int __Pyx_SetItemInt_Fast(PyObject *o, Py_ssize_t i, PyObje
         PyMappingMethods *mm = Py_TYPE(o)->tp_as_mapping;
         PySequenceMethods *sm = Py_TYPE(o)->tp_as_sequence;
         if (mm && mm->mp_ass_subscript) {
+            int r
```
Comment (scoder) [inline Cython/Utility/ObjectHandling.c]: Semicolon missing.

File: Cython/Utility/ObjectHandling.c [hunk 2]
```diff
@@ -432,8 +432,11 @@ static CYTHON_INLINE PyObject *__Pyx_GetItemInt_Fast(PyObject *o, Py_ssize_t i,
         PyMappingMethods *mm = Py_TYPE(o)->tp_as_mapping;
         PySequenceMethods *sm = Py_TYPE(o)->tp_as_sequence;
         if (mm && mm->mp_subscript) {
-            PyObject *key = PyInt_FromSsize_t(i);
-            return likely(key) ? mm->mp_subscript(o, key) : NULL;
+            PyObject r, *key = PyInt_FromSsize_t(i);
```
Comment (scoder) [inline Cython/Utility/ObjectHandling.c]: Pointer-star missing.

Comment (seberg) [reply Cython/Utility/ObjectHandling.c]: Heh, thanks... worst fixup ever

**Pull request merged**